index.d.ts

```diff
@@ -0,0 +1,486 @@
+    function setAccessToken(accessToken: string): void;
+
+    type Instruction = {
+        type: 'INSERT' | 'SET' | 'DELETE' | 'CLEAR' | 'LIST_SET' | 'LIST_INSERT' | 'LIST_ERASE' | 'LIST_CLEAR' | 'ADD_TYPE' | 'ADD_PROPERTIES'
+        object_type: string,
+        identity: string,
+        values: any | undefined
+        list_index: any | undefined
+        object_identity: any | undefined
+        new_identity: any | undefined,
+        property: any | undefined,
+        properties: any | undefined,
+        primary_key: string | undefined
+    }
+
+    class Adapter {
+        constructor(
+            local_path: string,
+            server_url: string,
+            admin_user: User,
+            regex: string,
+            change_callback: Function
+        )
+
+        /**
+         * Advance the to the next transaction indicating that you are done processing the current instructions for the given Realm.
+         * @param path the path for the Realm to advance
+         */
+        advance(path: string): void;
+        close(): void;
+        current(path: string): Array<Instruction>;
+        realmAtPath(path: string, realmID?: string, schema?: ObjectSchema[]): Realm
+    }
+}
+
+declare class Realm {
+    static defaultPath: string;
+
+    readonly path: string;
+    readonly readOnly: boolean;
+    readonly schema: Realm.ObjectSchema[];
+    readonly schemaVersion: number;
+
+    syncSession: Realm.Sync.Session;
+
+    /**
+     * @param  {string} path
+     * @param  {any} encryptionKey?
+     * @returns number
+     */
+    static schemaVersion(path: string, encryptionKey?: any): number;
+
+    /**
+     * @param  {Realm.Configuration} config?
+     */
+    constructor(config?: Realm.Configuration);
+
+    /**
+     * @returns void
+     */
+    close(): void;
+
+    /**
+     * @param  {string|Realm.ObjectType} type
+     * @param  {T&Realm.ObjectPropsType} properties
+     * @param  {boolean} update?
+     * @returns T
+     */
+    create<T>(type: string | Realm.ObjectType, properties: T & Realm.ObjectPropsType, update?: boolean): T;
+
+    /**
+     * @param  {Realm.Object|Realm.Object[]|Realm.List<any>|Realm.Results<any>|any} object
+     * @returns void
+     */
+    delete(object: Realm.Object | Realm.Object[] | Realm.List<any> | Realm.Results<any> | any): void;
+
+    /**
+     * @returns void
+     */
+    deleteAll(): void;
+
+    /**
+     * @param  {string|Realm.ObjectType} type
+     * @param  {number|string} key
+     * @returns T
+     */
+    objectForPrimaryKey<T>(type: string | Realm.ObjectType, key: number | string): T | void;
+
+    /**
+     * @param  {string|Realm.ObjectType} type
+     * @returns Realm
+     */
+    objects<T>(type: string | Realm.ObjectType): Realm.ObjectType & Realm.Results<T>;
+
+    /**
+     * @param  {string} name
+     * @param  {()=>void} callback
+     * @returns void
+     */
+    addListener(name: string, callback: () => void): void;
```

Comment:
`callback` should be `(sender: Realm, eventName: string) => void`. Or rather `(sender: Realm, event: 'change') => void` because that's all we support (which means `name: string` should be `event: 'change'`, too)